src/coreclr/nativeaot/BuildIntegration/Microsoft.NETCore.Native.targets

```diff
@@ -364,10 +365,10 @@ The .NET Foundation licenses this file to you under the MIT license.
         &quot;$(ObjCopyName)&quot; --strip-debug --strip-unneeded &quot;$(NativeBinary)&quot; &amp;&amp;
         &quot;$(ObjCopyName)&quot; --add-gnu-debuglink=&quot;$(NativeBinary)$(NativeSymbolExt)&quot; &quot;$(NativeBinary)&quot;" />
 
-    <Exec Condition="'$(StripSymbols)' == 'true' and '$(TargetOS)' == 'osx'"
+    <Exec Condition="'$(StripSymbols)' == 'true' and '$(TargetOS)' == 'osx' and '$(NativeLib)' != 'Static'"
```

Comment:
Does the equivalent non-osx target above work well static libraries? Should we have the same `'$(NativeLib)' != 'Static` condition on both places?

Reply:
Calling `dsymutil --flat` on static library `.a` gives this error on macOS:

```
EXEC : error : cannot parse the debug map for 'bin/Release/net8.0/osx-arm64/native/lib1.a':
The file was not recognized as a valid object file
```

On linux, objcopy handles static libs:

```
unstripped: 14.2M
stripped: 8.8M
```